default/scripting/buildings/TERRAFORM.focs.txt

```diff
@@ -16,20 +15,13 @@ BuildingType
             Planet environment = Adequate
         ]
     ]
-    EnqueueLocation = And [
-        Planet
-        OwnedBy empire = Source.Owner
-        Not Contains Building name = "BLD_TERRAFORM_REVERT"
-        Not Contains Building name = "BLD_GAIA_TRANS"
-        Not HasSpecial name = "GAIA_SPECIAL"
-        Not Enqueued type = Building name = "BLD_TERRAFORM_REVERT"
-        Not Enqueued type = Building name = "BLD_GAIA_TRANS"
-    ]
     effectsgroups = [
         EffectsGroup
             scope = And [
                 Object id = Source.PlanetID
                 Planet
+                Not HasSpecial name = "UNSTABLE_SPECIAL"
+                Not Planet type = LocalCandidate.NextBetterPlanetType
```

Comment:
Maybe I'm not understanding how that condition works right, but wouldn't that stop terraforming one step prior to reaching the optimal environment?

Reply:
Prevents the effect if the resulting type would be the same as current type.